test/put.sh

```diff
@@ -293,6 +293,33 @@ it_can_put_and_set_git_config() {
   mv ~/.gitconfig.orig ~/.gitconfig
 }
 
+it_can_put_and_force_the_push() {
+  local repo1=$(init_repo)
+
+  local src=$(mktemp -d $TMPDIR/put-src.XXXXXX)
+  local repo2=$src/repo
+  git clone $repo1 $repo2
+
+  local ref=$(make_commit $repo2)
+  make_commit $repo1
+  # cannot push to repo while it's checked out to a branch
+  git -C $repo1 checkout refs/heads/master
```

Comment:
Oh my bad - I misread and thought this commented line was setting up the force-push case. I see now that it's doing exactly what I described.